subsys/ipc/rpmsg_multi_instance/rpmsg_multi_instance.c

```diff
@@ -0,0 +1,453 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ipc/rpmsg_multi_instance.h>
+#include <zephyr.h>
+#include <device.h>
+#include <logging/log.h>
+
+#include <drivers/ipm.h>
+#include <openamp/open_amp.h>
+
+#define SHM_DEVICE_NAME          "sram0.shm"
+
+#define LOG_MODULE_NAME rpmsg_multi_instance
+LOG_MODULE_REGISTER(LOG_MODULE_NAME, CONFIG_RPMSG_MULTI_INSTANCE_LOG_LEVEL);
+
+#define RPMSG_VQ_0          (0) /* TX virtqueue queue index */
+#define RPMSG_VQ_1          (1) /* RX virtqueue queue index */
+#define VRING_COUNT         (2) /* Number of used vring buffers. */
+#define IPC_INSTANCE_COUNT  (CONFIG_RPMSG_MULTI_INSTANCES_NO) /* Total number of IPC instances.*/
+
+/* Private macros. */
+#define VRING_DESC_SIZEOF(num) ((num) * (sizeof(struct vring_desc)))
+#define VRING_AVAIL_SIZEOF(num) (sizeof(struct vring_avail) + \
+						((num) * sizeof(uint16_t)) + sizeof(uint16_t))
+#define VRING_ALIGN(size, align) (((size) + (align) - 1) & ~((align) - 1))
```

Comment:
You can just use `ROUND_DOWN` macro

Reply:
It seems to me that `ROUND_UP` macro should be used.